File: MAStaff-API/src/main/java/es/angelillo15/mast/api/cmd/SubCommand.java
package es.angelillo15.mast.api.cmd;

import org.bukkit.command.CommandSender;

public abstract class SubCommand {
    public abstract String getName();

    public abstract String getDescription();

    public abstract String getSyntax();

    public abstract String getPermission();

    public abstract void execute(CommandSender sender, String args[]);
}

File: MAStaff-API/src/main/java/es/angelillo15/mast/api/cmd/SubCommand.kt
package es.angelillo15.mast.api.cmd

import es.angelillo15.mast.api.cmd.sender.CommandSender

abstract class SubCommand {
    abstract val name: String?
    abstract val description: String?
    abstract val syntax: String?
    abstract val permission: String?
    abstract fun onCommand(sender: CommandSender?, label: String?, args: Array<String?>?)
}